Framework/DataObjects/src/ScanningWorkspaceBuilder.cpp

```diff
@@ -17,24 +17,14 @@ using namespace Mantid::Indexing;
 namespace Mantid {
 namespace DataObjects {
 
-ScanningWorkspaceBuilder::ScanningWorkspaceBuilder(const size_t nDetectors,
-                                                   const size_t nTimeIndexes,
-                                                   const size_t nBins)
-    : m_nDetectors(nDetectors), m_nTimeIndexes(nTimeIndexes), m_nBins(nBins),
-      m_histogram(BinEdges(nBins + 1, LinearGenerator(0.0, 1.0)),
+ScanningWorkspaceBuilder::ScanningWorkspaceBuilder(
+    const boost::shared_ptr<const Geometry::Instrument> &instrument,
+    const size_t nTimeIndexes, const size_t nBins)
+    : m_nDetectors(instrument->getNumberDetectors()),
+      m_nTimeIndexes(nTimeIndexes), m_nBins(nBins), m_instrument(instrument),
+      m_histogram(BinEdges(nBins + 1, LinearGenerator(1.0, 1.0)),
                   Counts(std::vector<double>(nBins, 0.0))),
```

Comment:
You can simply pass `nBins, 0.0`, `Counts` has the same constructors as `std::vector`.